### PR TITLE
[8.0][ADD] l10n_lu_ecdf: XML export of p&l, bs and chart of accounts, based on mis_builder

### PR DESCRIPTION
**Pull request by sbidoul**
Supersedes #13 
# Reports

This wizard allows to generate three types of financial reports :
- Profit & Loss (P&L)
- Balance Sheet (BS)
- Chart of Accounts (CA)

Profit & Loss and Balance Sheet can be generated in abbreviated version or not. The selected reports (max. 3) are written in a downloadable XML file for eCDF. 
# Templates

Five MIS Builder templates are used to generate the rapports : 
- Luxembourg Balance Sheet
- Luxembourg Balance Sheet (abbreviated)
- Luxembourg Profit & Loss
- Luxembourg Profit & Loss (abbreviated)
- Luxembourg Chart of Accounts
# View

ecdf_mis_report_gui

### CODE REVIEW

File: l10n_lu_ecdf/wizard/ecdf_report.py [hunk 1]
```diff
@@ -0,0 +1,746 @@
...
+                for report in data_curr:
+                    if not report['kpi_technical_name']:
+                        continue
+                    line_match = rexp.match(report['kpi_technical_name'])
+                    if line_match:
+                        ecdf_code = line_match.group('previous')
+                        record._append_num_field(form_data,
+                                                 ecdf_code,
+                                                 report['val'],
+                                                 zero=True)
```
Comment (sbidoul) [inline l10n_lu_ecdf/wizard/ecdf_report.py]: @ThomasBinsfeld I don't understand this `zero=True`. If I read correctly, it will always output 0.0 in this case, so what's the point?

**Comment by sbidoul** [comment]
@ThomasBinsfeld I did some tests and it looks pretty good. I only discovered one small issue with the file name being different than the file reference. Solved in https://github.com/OCA/l10n-luxemburg/pull/14/commits/6dbc909ad3a5ce004684a9123dbcac6883878d54

File: l10n_lu_ecdf/wizard/ecdf_report.py (hunk 1 above)
Comment (ThomasBinsfeld) [inline l10n_lu_ecdf/wizard/ecdf_report.py]: @sbidoul I use it when I populate the report with previous fiscal year's data. I need the value of the current year to know if it is AccountingNone or None (and ignore it) but the value to set in the report has to be 0.0.

Comment (sbidoul) [inline l10n_lu_ecdf/wizard/ecdf_report.py]: Hm. I find it more readable to put the test for None in the calling method like this: https://github.com/OCA/l10n-luxemburg/pull/14/commits/f3e745e58f3ab611deb63d282ec2ae564c31d1be. What do you think?

**Comment by sbidoul** [comment]
Most of the work has been done by @ThomasBinsfeld. I committed the latest fix after testing on real data.

:+1:  Great work.

**Comment by adrienpeiffer** [comment]
:+1: Thanks @ThomasBinsfeld !
